src/lib.rs

```diff
@@ -78,3 +78,62 @@ pub fn from_value<T: serde::de::DeserializeOwned>(value: JsValue) -> Result<T> {
 pub fn to_value<T: serde::ser::Serialize + ?Sized>(value: &T) -> Result<JsValue> {
     value.serialize(&Serializer::new())
 }
+
+/// A wrapper around a [`JsValue`] (or anything that can be cast into one)
+/// that makes it pass through serialization and deserialization unchanged.
+///
+/// # Example
+/// ```rust
+/// use serde_wasm_bindgen::{PreservedValue, to_value};
+/// #[derive(serde::Serialize)]
+/// struct MyStruct {
+///     int_field: i32,
+///     js_field: PreservedValue<JsValue>,
+/// }
+/// let big_array = js_sys::Int8Array::new_with_length(1000);
+/// let s = MyStruct {
+///     int_field: 5,
+///     js_field: PreservedValue(big_array.into()),
+/// };
+///
+/// // Will return a JsValue representing an object with two fields (`int_field` and `js_field`).
+/// // `js_field` will be an `Int8Array` pointing to the same underlying JavaScript object as `big_array`.
+/// to_value(&s);
+/// ```
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(bound = "T: JsCast")]
+pub struct PreservedValue<T: JsCast>(#[serde(with = "preserved_value")] pub T);
+
+// Some arbitrary string that no one will collide with unless they try.
+pub(crate) const PRESERVED_VALUE_MAGIC: &str = "1fc430ca-5b7f-4295-92de-33cf2b145d38";
+
+mod preserved_value {
```

Comment:
Hmm now that you added this module anyway, I think it's worth exposing it instead of the `PreservedValue` struct.

If exposed, it will allow anyone write something like

```rust
#[derive(Deserialize)]
struct Foo {
  a: i32,
  #[serde(with = "serde_wasm_bindgen::preserve")]
  b: JsValue,
}
```

instead of

```rust
#[derive(Deserialize)]
struct Foo {
  a: i32,
  b: PreservedValue<JsValue>,
}
```

This limits wrapping just to Serde actions, and to all the normal usages the value remains unwrapped and doesn't require doing `foo.b.0` everywhere instead of `foo.b`.

Of course, we could instead add usual helper traits `AsRef`, `AsMut`, `Deref`, `DerefMut`, `PartialEq<T>` etc. to the `PreservedValue<T>` struct to make it easier but making it a matter of Serde attribute seems more straightforward.

Reply:
Oh, cool, this does look much nicer to use!